internal/pkg/manifests/query/builder.go

```diff
@@ -0,0 +1,252 @@
+package query
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/thanos-community/thanos-operator/internal/pkg/manifests"
```

Comment:
same not again on ordering for consistency but driven entirely by my ocd and not a blocker :D